clients/java/src/main/java/io/camunda/zeebe/client/impl/response/ActivatedJobImpl.java

```diff
@@ -149,6 +154,11 @@ public String getTenantId() {
     return "";
   }
 
+  @Override
+  public Object getVariable(final String name) {
+    return getVariablesAsMap().get(name);
```

Comment:
On second thought, I see a potential issue with this. The JavaDoc does not specify what happens when no variable with that name exists. So, users might start to depend on the implementation, which is `null` or an NPE depending on the used map implementation (see [Map.get](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/util/Map.html#get(java.lang.Object)) documentation). 

🔧 It may be best to be explicit in our JavaDoc.

❌ We should also add a test case for the missing variable behavior. 

❓ This behavior may differ from a variable with `null` as value, so we must consider what behavior we like best here. What do you think is best?

Reply:
Nice point 👍 

If a variable as `null` value we should just simply return it. But in case the user provide a key that is not present in the variables map probably we should raise an error. So the user is aware of that. Of course this should be well explained in the javadoc. What do you think ?

Reply:
👍 I like it